src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -177,12 +182,42 @@ public function validate(mixed $entity, Constraint $constraint)
          * which is the same as the entity being validated, the criteria is
          * unique.
          */
-        if (!$result || (1 === \count($result) && current($result) === $entity)) {
+        if (!$result || (1 === \count($result) && current($result) === $value)) {
             return;
         }
 
-        $errorPath = $constraint->errorPath ?? $fields[0];
-        $invalidValue = $criteria[$errorPath] ?? $criteria[$fields[0]];
+        /* If a single entity matched the query criteria, which is the same as
+         * the entity being updated by validated object, the criteria is unique.
+         */
+        if (!$isEntity && !empty($constraint->identifierFieldNames) && 1 === \count($result)) {
```

Comment:
If `$isEntity` (so if you're using the constraint on a non-entity class), is the `identifierFieldNames` option required? If so, we should check earlier if it's missing and throw an exception. And then, remove the `empty()` call here. Unless I'm missing something, right now, it looks like if I match 1 result but I never set `identifierFieldNames`, I will always fail validation... and I can't think of when I would want that behavior.

Reply:
From PR description

> when the object update some entity, there is an option to provide the identifier field names to match that entity (optional)

If we just add new records then `identifierFieldNames` option is not required.
Example 3 from PR description updates entity. In this case, we need to set `identifierFieldNames` or UniqueEntity validator would fail by matching the entity that is updated.

The purpose of this condition is to check if:
- it has dealt with a non-entity and
- `identifierFieldNames` option is set and
- found one non-unique record

Because the record found, may be the same one we just updating/validating.

Reply:
Setting `identifierFieldNames` option without setting `entityClass` is pointless. 
I could add an exception early:
`throw new ConstraintDefinitionException('entityClass needs to be specified when the identifierFieldNames was set');`